tests/userRegistration.test.ts

```diff
@@ -59,11 +68,23 @@ describe("User Registration", () => {
 
       await sendRegisterPostRequest(data).expect(STATUS_CODES.CREATED);
     });
+
+    it("should send a confirmation email", async () => {
+      const data = { email, password };
+      const userToCreate = { email, password } as ResolvedValue<PrismaUser>;
+
+      prismaMock.user.create.mockResolvedValue(userToCreate);
+
+      await sendRegisterPostRequest(data);
+
+      expect(sendConfirmationEmail).toHaveBeenCalled();
+      expect(sendConfirmationEmail).toHaveBeenCalledWith(email);
```

Comment:
expect(sendConfirmationEmail).toHaveBeenCalledWith(email); is enough because with that you are testing that it's called and with a specific email. So you don't need this `expect(sendConfirmationEmail).toHaveBeenCalled();`

Reply:
Fixed.